### PR TITLE
History

### LINKED ISSUE #17: Exploring History
https://ucsdlib.github.io/git-novice/05-history/

### PR DESCRIPTION
**Pull request by mcritchlow**
Closes #17

### CODE REVIEW

File: _episodes/04-history.md [hunk 1]
```diff
@@ -20,41 +20,50 @@ As we saw in the previous lesson, we can refer to commits by their
...
+--- a/README.md
++++ b/README.md
+@@ -8,3 +8,4 @@ Image credits:
+ - A Xmas shopper (http://digitalcollections.nypl.org/items/510d47e1-43b5-a3d9-e040-e00a18064a99). Courtesy New York Public Library.
+ 
+ The images are in the `images/` subdirectory of this repository.
++If you would like, you can also put pictures of dogs and hexagonal software logos here.
+
 ~~~
 {: .output}
```
Comment (dunn) [inline _episodes/04-history.md]: Everything above here is already covered in the previous section, I think.

**Review by dunn — CHANGES_REQUESTED** [review]

**Comment by dunn** [comment]
There are still old names and files referenced in Recovering Older Versions of a File and beyond.

**Comment by dunn** [comment]
Also, should we discuss `git show`?

File: _episodes/04-history.md (hunk 1 above)
Comment (mcritchlow) [reply _episodes/04-history.md]: true. I think the repetition is intentional from what I recall being told about some of the thinking behind this lessons. Or the assumption that ppl might have gone on a break, and are starting 'fresh' again. You're not wrong, I just didn't think to mess with that.

Comment (dunn) [reply _episodes/04-history.md]: Chrissy can be the tie-breaker, but I think we should try dropping it.

Comment (dunn) [reply _episodes/04-history.md]: I mean @chrissyrissmeyer

File: _episodes/04-history.md [hunk 2]
```diff
@@ -128,23 +149,26 @@ and "unique" really does mean unique:
 every change to any set of files on any computer
```
Comment (dunn) [inline _episodes/04-history.md]: This sentence is really weird; it makes it sound like the same commit will have a different hash on each computer.

**Comment by dunn** [comment]
Can we briefly discuss reflog?

File: _episodes/04-history.md [hunk 3]
```diff
@@ -250,25 +294,21 @@ $ git status
...
-Changes to be committed:
-  (use "git reset HEAD <file>..." to unstage)
-# Changes not staged for commit:
-#   (use "git add <file>..." to update what will be committed)
-#   (use "git checkout -- <file>..." to discard changes in working directory)
-#
-#	modified:   mars.txt
-#
-no changes added to commit (use "git add" and/or "git commit -a")
+On branch master
```
Comment (dunn) [inline _episodes/04-history.md:270]: Won't this say something like 'detached HEAD state', instead of `master`?

**Comment by mcritchlow** [comment]
i dunno about `reflog`.. it seems like a 'next level' thing and less novice when there's already a lot of new to process. but it may just be because i've never included it before.

File: _episodes/04-history.md (hunk 3 above)
Comment (mcritchlow) [reply _episodes/04-history.md:270]: We did a checkout of README, not a checkout of the commit w/ no referenced file. I can go back and check my history, but I believe this is right.

File: _episodes/04-history.md (hunk 1 above)
Comment (mcritchlow) [reply _episodes/04-history.md]: Yeah that's fine. I think the nice thing here is the transition from using HEAD to integrating the `~1` syntax.

**Comment by dunn** [comment]
Do we discuss rebasing at any point? If so I think reflog is important to make people feel like they aren't risking everything.  Otherwise it might not be essential.

File: _episodes/04-history.md (hunk 3 above)
Comment (dunn) [reply _episodes/04-history.md:270]: Oh, I see.

**Comment by mcritchlow** [comment]
No, there's nothing about rebasing in this class.

File: _episodes/04-history.md (hunk 1 above)
Comment (dunn) [reply _episodes/04-history.md]: We could look into introducing HEAD in the previous section.

Comment (dunn) [reply _episodes/04-history.md]: But at that point we might have to split up section 3 into two again.

Comment (mcritchlow) [reply _episodes/04-history.md]: Yeah, I guess my vote is to leave it alone. Glossing over it quickly isn't an issue I think as it's really only a couple commands, and it's less potential refactoring or touching of other lessons. This is mostly out of laziness at this point, I admit..

Comment (dunn) [reply _episodes/04-history.md]: D:

Comment (chrissyrissmeyer) [reply _episodes/04-history.md]: My general feeling is that it is good to remind folks of commands that they should be using regularly (e.g., git status) or at certain points, but not explain it again unless there is something in the new section that it amplifies or clarifies.

Comment (mcritchlow) [reply _episodes/04-history.md]: Ok, I removed the initial git diff check. I still think the new line needs to be added to the README to actually demo the diff's, so I've left that in. The previous section does introduce `diff` and `diff --staged`, so I think this now carries on from where that leaves off (diff'ing against previous commits. Is this better, or was there more you all were hoping to drop in this section?

Comment (dunn) [reply _episodes/04-history.md]: We don't need the new line to show diffs like `HEAD~1..HEAD`, and we've already demonstrated diffing between HEAD and the uncommitted changes.

File: _episodes/04-history.md [hunk 4]
```diff
@@ -20,95 +20,93 @@ As we saw in the previous lesson, we can refer to commits by their
 identifiers.  You can refer to the _most recent commit_ of the working
 directory by using the identifier `HEAD`.
 
-We've been adding one line at a time to `mars.txt`, so it's easy to track our
+We've been adding one line at a time to `README.md`, so it's easy to track our
 progress by looking, so let's do that using our `HEAD`s.  Before we start,
-let's make a change to `mars.txt`.
+let's make a change to `README.md`.
```
Comment (dunn) [inline _episodes/04-history.md]: 'another change'

Comment (mcritchlow) [reply _episodes/04-history.md]: done

**Review by dunn — APPROVED** [review]

**Pull request merged**